crates/net/network/src/network.rs

```diff
@@ -219,19 +222,23 @@ impl PeersInfo for NetworkHandle {
     }
 
     fn local_node_record(&self) -> NodeRecord {
-        let id = *self.peer_id();
-        let mut socket_addr = *self.inner.listener_address.lock();
-
-        if socket_addr.ip().is_unspecified() {
-            // zero address is invalid
-            if socket_addr.ip().is_ipv4() {
-                socket_addr.set_ip(std::net::IpAddr::V4(std::net::Ipv4Addr::LOCALHOST));
-            } else {
-                socket_addr.set_ip(std::net::IpAddr::V6(std::net::Ipv6Addr::LOCALHOST));
+        if let Some(discv4) = &self.inner.discv4 {
+            discv4.node_record()
+        } else {
+            let id = *self.peer_id();
+            let mut socket_addr = *self.inner.listener_address.lock();
+
+            if socket_addr.ip().is_unspecified() {
+                // zero address is invalid
+                if socket_addr.ip().is_ipv4() {
+                    socket_addr.set_ip(std::net::IpAddr::V4(std::net::Ipv4Addr::LOCALHOST));
+                } else {
+                    socket_addr.set_ip(std::net::IpAddr::V6(std::net::Ipv6Addr::LOCALHOST));
+                }
```

Comment:
not sure about this behaviour, I think the node record should advertise no ip when listening on all interfaces, not localhost. it could be WAN-reachable, just still waiting on nat or discv4 to resolve its WAN-reachable address. (@mattsse )

Reply:
it should include an ip, it just cant be 0.0.0.0, this is also how geth behaves:

```text
cast rpc admin_nodeInfo --rpc-url http://localhost:8548 | jq
{
  "id": "7f88ffc891d168a87d389b9a864dce5d5681ed36c86319b262de37d4eca22574",
  "name": "Geth/v1.13.11-stable/darwin-arm64/go1.21.6",
  "enode": "enode://3aa6582fd0a5e188c449927d1e870a717312cc45ad76216744d8852d1cb2dc9577d54421203bf90c9960bee5fec4150906e757e3cfb7f66c748c02666247bce1@127.0.0.1:30303",
```